### PR TITLE
SQL Phone

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by setoy** [comment]
This looks neat. Any chance this is gonna be implemented? Nice work!

**Comment by CarlosHingel** [comment]
@GetSomePanda Is this the same version that you released on altislife forum? Just asking because that version has two little bugs hehe

1 - When you click "answer" button, the phone does not select the person that you want to answer... You still need to find the person on the list.

2 - When you click some buttons to send messages(private message, police, medics, admin and etc) another button of the phone will also flash like you had clicked but nothing happens the button that you've clicked is the one that works like it should, in other words, it's just a small visual glitch of two buttons flashing when you a click one button.

**Comment by DarkSilencerZA** [comment]
Yes this is the bugged version where "Reply" does not work.

**Comment by GetSomePanda** [comment]
Will fix the reply button.

**Comment by DarkSilencerZA** [comment]
extDB2: Version: 71
extDB2: https://github.com/Torndeco/extDB2
extDB2: Windows Version
Message: All development for extDB2 is done on a Linux Dedicated Server
Message: If you would like to Donate to extDB2 Development
Message: https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=2SUEFTGABTAM2
Message: Also leave a message if there is any particular feature you would like to see added.
Message: Thanks for all the people that have donated.
Message: Torndeco: 20/02/15


extDB2: Found extdb-conf.ini
extDB2: Detected 8 Cores, Setting up 6 Worker Threads


[16:02:04:505114 +01:00] [Thread 6608] extDB2: Database Type: MySQL
[16:02:04:510803 +01:00] [Thread 6608] extDB2: Database Session Pool Started
[16:02:04:511122 +01:00] [Thread 6608] extDB2: SQL_RAW_V2: Initialized: ADD_QUOTES True
[16:19:38:803822 +01:00] [Thread 6500] extDB2: SQL_RAW_V2: Error StatementException: MySQL: [MySQL]: [Comment]: mysql_stmt_prepare error	[mysql_stmt_error]: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''"[[`[JNA] Meto`,`corp`,[1,[2016,8,14,16,18]]],[`[JNA] Meto`,`hey u guys corp or' at line 1	[mysql_stmt_errno]: 1064	[mysql_stmt_sqlstate]: 42000	[statemnt]: UPDATE cellphone SET messages='"[[`[JNA] Meto`,`corp`,[1,[2016,8,14,16,18]]],[`[JNA] Meto`,`hey u

**Comment by DarkSilencerZA** [comment]
The above is the start of a very LONG and HUGE file full of errors.

**Comment by Jawshy** [comment]
> @DarkSilencerZA, do you even [markdown](https://guides.github.com/features/mastering-markdown/)?

**Comment by DarkSilencerZA** [comment]
@Jawshy Why u do dis :(

**Comment by tkcjesse** [comment]
More like pastebin...

**Comment by Jawshy** [comment]
Or just upload directly to GitHub.

**Comment by DarkSilencerZA** [comment]
It spams so much that pastebin can't take the size of it. That's how bad it is..

**Comment by DarkSilencerZA** [comment]
My god this phone is buggy :/

**Comment by BoGuu** [comment]
WIP

File: altislife.sql [hunk 1]
```diff
@@ -204,6 +204,22 @@ CREATE TABLE IF NOT EXISTS `wanted` (
...
 -- --------------------------------------------------------
 
+--
+-- Table structure for table `cellphone`
+-- Only run this if you already have a database made on your server.
+-- Made by GetSomePanda / Panda
+--
+
+CREATE TABLE IF NOT EXISTS `cellphone` (
+  `pid` varchar(64) NOT NULL,
```
Comment (Jawshy) [inline altislife.sql]: Why `64` and not `32` like the others?

File: altislife.sql [hunk 2]
```diff
@@ -204,6 +204,22 @@ CREATE TABLE IF NOT EXISTS `wanted` (
...
+  `playerName` varchar(32) NOT NULL,
+  `messages` text NOT NULL,
+  `insert_time` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  PRIMARY KEY (`pid`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```
Comment (Jawshy) [inline altislife.sql]: `utf8mb4`*

File: altislife.sql (hunk 1 above)
Comment (BoGuu) [inline altislife.sql]: or just 17, like the others should be

**Comment by Jawshy** [comment]
The following error in expression was output to the client/local RPT log file when clicking reply to a received text message: 

```
 0:27:18 Error in expression <l [3004,0];
} 
else
{
lbSetCurSel [3004,_lbNum];
};>
 0:27:18   Error position: <_lbNum];
};>
 0:27:18   Error Undefined variable in expression: _lbnum
 0:27:18 File mpmissions\__CUR_MP.Altis\core\pmenu\fn_cellphone.sqf, line 58
```

[Altis_Life.Altis/core/pmenu/fn_cellphone.sqf, line 58](https://github.com/GetSomePanda/Framework/blob/471f1806afd462f052d6a7aa29edd7b634b95401/Altis_Life.Altis/core/pmenu/fn_cellphone.sqf#L58)

Police and text admin text messages are not being stored. 

My extDB2 logs do not have any of @DarkSilencerZA's reported errors.

File: altislife.sql (hunk 1 above)
Comment (Jawshy) [inline altislife.sql]: Your wish is my command. https://github.com/AsYetUntitled/Framework/pull/72

**Comment by DarkSilencerZA** [comment]
The reported errors were tested with @tkcjesse  - they appeared in his logs too.

**Comment by tkcjesse** [comment]
@GetSomePanda You should really create a sql procedure to delete old messages.

**Comment by sargken** [comment]
@GetSomePanda Status

**Comment by fresqo** [comment]
@tkcjesse @GetSomePanda yeah the database can get filled up very quickly. I use to rpt log all the messages instead so messages were only used by admins etc for claims.

**Comment by DarkSilencerZA** [comment]
` 9:41:00 Error in expression <C_Life];
} else {
[getPlayerUid player, _messages] remoteExecCall ["DB_fnc_saveC>
 9:41:00   Error position: <_messages] remoteExecCall ["DB_fnc_saveC>
 9:41:00   Error Undefined variable in expression: _messages
 9:41:00 File mpmissions\__CUR_MP.Altis\core\session\*fn_updateRequest.sqf, line 57`

**Comment by MJRamon** [comment]
Just keep last X messages per user or flush them at server start. Any progress, BTW?

**Comment by mur202** [comment]
does not work with the new extdb3

**Comment by Jawshy** [comment]
@mur202, see [/life_server/Functions/MySQL/fn_getRealTime.sqf, line 8](https://github.com/GetSomePanda/Framework/blob/c168133c89d6881e2d6dbb42c97f66bea4a18531/life_server/Functions/MySQL/fn_getRealTime.sqf#L8).

**Comment by Bagsi** [comment]
It works on life 5.0.0 ?

**Comment by RosieCode95** [comment]
closing due to stale pr and incompatibility, i may take the time to update and resubmit after discussion of its place in this repo

**Pull request closed without merging**